src/sentry/models/environment.py

```diff
@@ -88,11 +88,16 @@ def get_or_create(cls, project, name):
         return env
 
     def add_project(self, project):
-        try:
-            with transaction.atomic():
-                EnvironmentProject.objects.create(project=project, environment=self)
-        except IntegrityError:
-            pass
+        cache_key = 'envproj:c:%s:%s' % (self.id, project.id)
+
+        envproj = cache.get(cache_key)
+        if envproj is None:
```

Comment:
Just a nit, since you're not doing anything with this var, you can just `if not cache.get(cache_key):`